xarray/core/duck_array_ops.py

```diff
@@ -190,11 +191,7 @@ def lazy_array_equiv(arr1, arr2):
     arr2 = asarray(arr2)
     if arr1.shape != arr2.shape:
         return False
-    if (
-        dask_array
-        and isinstance(arr1, dask_array_type)
-        and isinstance(arr2, dask_array_type)
-    ):
+    if dask_array and is_duck_dask_array(arr1) and is_duck_dask_array(arr2):
         # GH3068
         if arr1.name == arr2.name:
```

Comment:
```suggestion
        if dask.base.tokenize(arr1) == dask.base.tokenize(arr2):
```

This should work as long as `pint` implements `__dask_tokenize__`

It seems like `return (dask_array.name, self.units)` or something would work.